Add Models to Bookmarks app

pavan-testpress opened the pull request:
- Add Bookmarksapp to Installed apps
- Add Bookmarks Model
- Add Folders Model
- Add django_utils Pipfile
- Migrate the Models
- Remove Unused imports to follow flake8 guidelines
- Add Bookmarks, Folders Models to Admin site

pavan-testpress committed:
Add Models to Bookmarks app

- Add Bookmarksapp to Installed apps
- Add Bookmarks Model
- Add Folders Model
- Add django_utils Pipfile
- Migrate the Models
- Remove Unused imports to follow flake8 guidelines
- Add Bookmarks, Folders Models to Admin site

waranlogesh commented on DjangoProject/bookmarksapp/admin.py:
```diff
@@ -1,3 +1,8 @@
 
+from .models import Bookmarks, Folders
+
 # Register your models here.
```
Remove the auto-generated comments

pavan-testpress replied:
Removed the auto-generated comments

waranlogesh commented on DjangoProject/bookmarksapp/models.py:
```diff
@@ -1,3 +1,30 @@
+
+from authenticationapp.models import MyUser
 
 # Create your models here.
```
Remove it

pavan-testpress replied:
Removed the auto-generated comments

waranlogesh commented on DjangoProject/bookmarksapp/models.py:
```diff
@@ -1,3 +1,30 @@
+
+
+class Folders(TimeStampedModel):
+    name = models.CharField(max_length=50, unique=True)
```
`unique = True`? So what if another user whats to create the same folder name.

Consider `user1` has created a folder with the name `google` and `user2` tries to create a folder with the same name `google` and your design won't allow that.

Ideally, it should be allowed and uniqueness for folder name should be check with respect to the user.

pavan-testpress replied:
I removed unique constraint. will check in views while saving.

waranlogesh commented on DjangoProject/bookmarksapp/models.py:
```diff
@@ -1,3 +1,30 @@
+    folder = models.ForeignKey(Folders, on_delete=models.SET_NULL, related_name='folder', null=True)
+    name = models.CharField(max_length=50, unique=True)
+    url = models.URLField(max_length=200, unique=True)
+    user = models.ForeignKey(MyUser, on_delete=models.CASCADE, default=1)
```
Why do u have a user field here? Field name should be verbose. If it's for storing the user who created the bookmark it should be `created_by`. Why is `default=1`?

pavan-testpress replied:
- changed user to created_by.
- added null=True
- remove default = 1

pavan-testpress committed:
Change Bookmarks, Folder models

- Remove Unique Constraint on names & urls fileds on both models
- Remove Auto generated comments
- Change field name user to created_by(verbose)

pavan-testpress committed:
Migrate as a single file

pavan-testpress committed:
Add migration file
